src/bytes.rs

```diff
@@ -51,16 +57,40 @@ const _: () = {
 };
 
 extern "C" fn noop(_: *const (), _: usize) {}
+
 impl<'a> Bytes<'a> {
     /// Constructs an empty slice.
     ///
     /// Since [`Bytes`] are immutable slices, this mostly serves to create default values.
     pub const fn empty() -> Self {
         Self::from_static([].as_slice())
     }
+    const fn is_interned(&self) -> bool {
+        (unsafe {
```

Comment:
```suggestion
        // read the last bit of `.vtable` (parity bit on little-endian architectures)
        (unsafe {
```
        
To avoid the type stutter, we could even define a helper function:

```rs
const unsafe fn read_last_byte_of(ptr: &T) -> bool {
    ptr::addr_of!(*ptr).cast::<u8>().add(mem::size_of::<T>()).read()
}
```

so as, to, here:

```rs
        unsafe { read_last_byte_of(&self.vtable) } & 0b1 != 0b0
```

but I can understand if that feels excessive (although it will be useful later on, I think)